### PR TITLE
Deprecate the Krew plugin

### PR DESCRIPTION
**Pull request by dvaldivia**
We are deprecating the krew plugin in favor of focusing on the `kustomize` and `helm` approach of installing the operator.

### CODE REVIEW

**Comment by shtripat** [comment]
There is an enterprise test using this for expand tenant feature test. We may need to change that.

**Review by cniackz — DISMISSED** [review]
LGTM

**Comment by cniackz** [comment]
Tested and deployed the tenant with this compiled version. It shouldn't break functionality, although I agree with @shtripat  that tests should catch up with the changes, as should the documentation. @ravindk89, please be aware.

Screenshot 2024-03-28 at 8 32 47 AM

**Comment by cniackz** [comment]
Oh I see, tests are updated and regarding `common.sh` these changes looks good to me:

Screenshot 2024-03-28 at 8 37 31 AM

**Comment by allanrogerr** [comment]
@dvaldivia @cniackz 
This is a breaking change to our users' workflows. Therefore how will this be announced, giving users sufficient time to change their workflows?

**Review by allanrogerr — CHANGES_REQUESTED** [review]
@dvaldivia @cniackz
This is a breaking change to our users' workflows. Therefore how will this be announced, giving users sufficient time to change their workflows?

**Comment by ravindk89** [comment]
@dvaldivia @cniackz if I may suggest:

- Hold this PR for now
- Open a fresh PR that marks all `kubectl minio` commands as deprecated and returns a non-blocking warning on each run (this may break user workflows who parse output in some way)
- Docs will warn that we are deprecating `kubectl minio` route of installation and focus on `kubectl apply -k github.com/minio/operator` and the equivalent for Tenants
- Merge this PR for the subsequent Operator release

With this, we would

- Publish 5.0.15 with a deprecation warning
- Publish 5.0.16 with the deprecation

Users do not always track latest stable that closely, and can stay on 5.0.15 until they have readied their systems to move to Kustomize.

Thoughts?

File: Makefile [hunk 1]
```diff
@@ -79,17 +78,6 @@ regen-crd-docs:
 	@${GOPATH}/bin/crd-ref-docs --source-path=./pkg/apis/sts.min.io/v1alpha1  --config=docs/templates/config.yaml --renderer=asciidoctor --output-path=docs/policybinding_crd.adoc --templates-dir=docs/templates/asciidoctor/
 	@${GOPATH}/bin/crd-ref-docs --source-path=./pkg/apis/job.min.io/v1alpha1  --config=docs/templates/config.yaml --renderer=asciidoctor --output-path=docs/job_crd.adoc --templates-dir=docs/templates/asciidoctor/
 
-plugin: regen-crd
```
Comment (allanrogerr) [inline Makefile:82]: We dont need krew to build and run kubectl-minio . Why is it being removed as well?

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
> This is a breaking change to our users' workflows. Therefore how will this be announced, giving users sufficient time to change their workflows?

There is no such workflow

**Comment by harshavardhana** [comment]
> @dvaldivia @cniackz if I may suggest:
> 
> * Hold this PR for now
> * Open a fresh PR that marks all `kubectl minio` commands as deprecated and returns a non-blocking warning on each run (this may break user workflows who parse output in some way)
> * Docs will warn that we are deprecating `kubectl minio` route of installation and focus on `kubectl apply -k github.com/minio/operator` and the equivalent for Tenants
> * Merge this PR for the subsequent Operator release
> 
> With this, we would
> 
> * Publish 5.0.15 with a deprecation warning
> * Publish 5.0.16 with the deprecation
> 
> Users do not always track latest stable that closely, and can stay on 5.0.15 until they have readied their systems to move to Kustomize.
> 
> Thoughts?

No we move to v6.0.0 - we break this and bump the major version.

**Comment by feorlen** [comment]
What is the timeline for this? It's a major change to how docs describes the k8s install process. We do have some preliminary docs for kustomize in GitHub, but multiple pages in the web docs need rework to feature that instead of krew.

**Review by allanrogerr — CHANGES_REQUESTED** [review]
Consider the changes to goreleaser made here: https://github.com/minio/operator/pull/2062

**Comment by cniackz** [comment]
I will solve the conflict:

Screenshot 2024-04-17 at 5 32 10 PM

**Review by cniackz — APPROVED** [review]

**Comment by cniackz** [comment]
The conflict has been resolved, and now I'm awaiting the completion of tests. Once they pass, I'll request the team to review again, and then proceed with the merge.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**